sycl/test-e2e/Matrix/joint_matrix_tensorcores_sm70.cpp

```diff
@@ -80,12 +80,23 @@ int main() {
     test<const half, const half, float, SUB_TILES_M, SUB_TILES_K, SUB_TILES_N,
          32, 16, 8>(Q);
 
+    // test different layout combinations for one case
+
+    test<const half, const half, float, SUB_TILES_M, SUB_TILES_K, SUB_TILES_N,
+         32, 16, 8, layout::row_major, layout::row_major, layout::col_major>(Q);
```

Comment:
I'm wondering, C row major is tested with all combinations of A and B layout, while C col major is tested only with A and B row major, but no col/row and row/col and col/col combinations tested. Is it by intention?

Reply:
Yes. The implementation of loading/storing matrices of type C does not behave differently depending on the layouts of type A/B matrices. I know this for certain the the cuda impl, (Aside: and other backends are likely to have the same situation; but this is irrelevant here). Therefore testing of such additional combinations would be redundant.

However the implementation does depend on the pairing of A/B layouts, and therefore it is meaningful to test explicitly:

- A=row_major, B=row_major
- A=col_major, B=row_major
- A=row_major, B=col_major
- A=col_major, B=col_major